tls/s2n_connection.c

```diff
@@ -1288,14 +1288,13 @@ int s2n_connection_send_stuffer(struct s2n_stuffer *stuffer, struct s2n_connecti
     S2N_ERROR_IF(s2n_stuffer_data_available(stuffer) < len, S2N_ERR_STUFFER_OUT_OF_DATA);
 
     int w = 0;
-    do {
-        errno = 0;
-        w = conn->send(conn->send_io_context, stuffer->blob.data + stuffer->read_cursor, len);
-        if (w < 0 && errno == EPIPE) {
-            conn->write_fd_broken = 1;
-        }
-        S2N_ERROR_IF(w < 0 && errno != EINTR, S2N_ERR_SEND_STUFFER_TO_CONN);
-    } while (w < 0);
+    errno = 0;
+    S2N_IO_RETRY_EINTR(w,
+            conn->send(conn->send_io_context, stuffer->blob.data + stuffer->read_cursor, len));
+    if (w < 0 && errno == EPIPE) {
+        conn->write_fd_broken = 1;
+    }
+    POSIX_ENSURE(w >= 0, S2N_ERR_SEND_STUFFER_TO_CONN);
```

Comment:
Correct, I'm very specifically not modifying the behavior of the existing, non-ktls IO in this PR. There might be an argument for updating their behavior, but it's out of the scope of this PR.

I don't think the old behavior is really a big problem. It just moves the error handling higher in the stack. Which sure I don't love because the further you get from the original syscall the more likely you make a mistake with errno, but isn't automatically wrong.